Clinical Key.js

```diff
@@ -0,0 +1,190 @@
+Replicate by going to a content page (/content/book/...) and then going to a broswing page (/browse/book/...).
+URL remains the page of the previous page (content page) and doc.location.href is the current page (as it should be).
+
+Hence, I never use url and change its content to doc.location.href in detectWeb, the only function that uses the url
+*/
+
+function detectWeb(doc, url) {
+	//see NB above for explanation
+	url = doc.location.href;
+
+	//contentType depends on url, which is present, but rest of site is loaded via ajax (I think)
+	//monitor dom and reset if changes
+	var jsession = doc.getElementById('jsessionid');
+	Z.debug(jsession.value);
+	if(jsession){
+		Zotero.monitorDOMChanges(jsession, {attributes:true});
+		if(!jsession.value){
+			return;
+		}
+	}
+
+	var contentType;
+	if(url.includes("/content/book/") && !url.includes("login?")){
+		contentType = "bookSection";
+	}
+	else if (url.includes('/browse/book/') && !url.includes("login?")){
+		contentType = "book";
+	}
+
+	//contentType not set
+	if(!contentType){
+		return;
+	}
+	return contentType;
+}
+
+function doWeb(doc, url){
+	var contentType = detectWeb(doc, url);
+	//if contentType null or not equal to bookSection or book
+	if(!contentType || !(contentType == 'bookSection' || contentType == 'book')){
+		return;
+	}
+
+	//if book section
+	if (contentType == 'bookSection'){
+		var newItem = new Zotero.Item(contentType);
+		newItem = scrapeBookSection(doc, newItem);
+		//pdf (if present)
+		var pdfUrl = doc.getElementsByClassName('x-pdf')[0].href;
+		if(pdfUrl){
+			newItem.attachments.push({
+				url:pdfUrl,
+				title:"Book Section PDF",
+				mimeType:"application/pdf"
+			});
+		}
+		//populate common attributes
+		//url, see NB above for explanation as to why url is NOT used
+		newItem.url = doc.location.href;
+		newItem.complete();
+	}
+	//if book, use ISBN translator
+	//borrowed from amazon translator
+	else if (contentType == 'book'){
+		var isbn = getElementsByAttr(doc, "data-metadata-isbn", "")[0].getAttribute('data-metadata-isbn');
+		isbn = ZU.cleanISBN(isbn);
+		//use search translator to get metadata from isbn
+		var search = Zotero.loadTranslator("search");
+		//set translators and search
+		search.setHandler("translators", function(obj, translators) {
+			search.setTranslator(translators);
+			search.setHandler("itemDone", function(obj, lookupItem) {
+				newItem=lookupItem;
+				//update ISBN
+				newItem.ISBN = ZU.cleanISBN(isbn);
+				//update url, see NB for rationale why url not used
+				newItem.url = doc.location.href;
+			});
+			search.translate();
+		});
+		//no need to override error handler
+		//save item
+		search.setHandler("done", function() {
+			newItem.complete();
+		});
+		search.setSearch({ ISBN: isbn });
+		search.getTranslators();
+	}
+}
+
+function scrapeBookSection(doc, item){
+	//book title
+	var bookTitle = getElementsByAttr(doc, 'data-once-text','XocsCtrl.title')[0].innerHTML;
```

Comment:
generally, see if you can't replace this custom functin by `ZU.xpathText` (where you're getting the innerHTML) or `ZU.xpath` (where you don't). Maybe I'm wrong here, but if you can it would make maintaining this much easier and your custom function isn't super efficient, either.